File: common/src/main/java/dev/chasem/cobblemonextras/config/CobblemonExtrasConfig.java
package dev.chasem.cobblemonextras.config;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.annotations.SerializedName;


public class CobblemonExtrasConfig {

    public static Gson GSON = new GsonBuilder()
            .disableHtmlEscaping()
            .setPrettyPrinting()
            .create();
    @SerializedName("permissionlevels") public PermissionLevels permissionLevels = new PermissionLevels();
    public ShowcaseConfig showcase = new ShowcaseConfig();
    public class PermissionLevels {
        @SerializedName("command.poketrade") public int COMMAND_POKETRADE_PERMISSION_LEVEL = 2;
        @SerializedName("command.pokesee") public int COMMAND_POKESEE_PERMISSION_LEVEL = 2;
        @SerializedName("command.pokeseeother") public int COMMAND_POKESEEOTHER_PERMISSION_LEVEL = 2;
        @SerializedName("command.pc") public int COMMAND_PC_PERMISSION_LEVEL = 2;
        @SerializedName("command.pokeshout") public int COMMAND_POKESHOUT_PERMISSION_LEVEL = 2;
        @SerializedName("command.itemshout") public int COMMAND_ITEMSHOUT_PERMISSION_LEVEL = 2;
        @SerializedName("command.compsee") public int COMMAND_COMPSEE_PERMISSION_LEVEL = 2;
        @SerializedName("command.compseeother") public int COMMAND_COMPSEEOTHER_PERMISSION_LEVEL = 2;
        @SerializedName("command.pokebattle") public int COMMAND_POKEBATTLE_PERMISSION_LEVEL = 3;
        @SerializedName("command.battle") public int COMMAND_BATTLE_PERMISSION_LEVEL = 2;
        @SerializedName("command.comptake") public int COMMAND_COMPTAKE_PERMISSION_LEVEL = 2;
        @SerializedName("command.pokeivs") public int COMMAND_POKEIVS_PERMISSION_LEVEL = 2;
        @SerializedName("command.emptybox") public int COMMAND_EMPTYBOX_PERMISSION_LEVEL = 2;
        @SerializedName("command.pokeshoutall") public int COMMAND_POKESHOUT_ALL_PERMISSION_LEVEL = 2;
        @SerializedName("command.pokeodds") public int COMMAND_POKEODDS_PERMISSION_LEVEL = 3;
        @SerializedName("command.pokekill") public int COMMAND_POKEKILL_PERMISSION_LEVEL = 3;
    }

    public class ShowcaseConfig {
        public boolean isShowcaseEnabled = true;
        public String apiSecret = "To start using showcase, please goto https://cobblemonextras.com/showcase";

        public int syncIntervalMinutes = 5;
        public boolean debug = false;
    }
}

File: common/src/main/kotlin/dev/chasem/cobblemonextras/config/CobblemonExtrasConfig.kt
package dev.chasem.cobblemonextras.config

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

class CobblemonExtrasConfig {
    @SerializedName("permissionlevels")
    var permissionLevels: PermissionLevels = PermissionLevels()
    var showcase: ShowcaseConfig = ShowcaseConfig()
    @SerializedName("customModels")
    var customModels: CustomModelsConfig = CustomModelsConfig()


    inner class PermissionLevels {
        @SerializedName("command.poketrade")
        var COMMAND_POKETRADE_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pokesee")
        var COMMAND_POKESEE_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pokeseeother")
        var COMMAND_POKESEEOTHER_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pc")
        var COMMAND_PC_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pokeshout")
        var COMMAND_POKESHOUT_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.itemshout")
        var COMMAND_ITEMSHOUT_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.compsee")
        var COMMAND_COMPSEE_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.compseeother")
        var COMMAND_COMPSEEOTHER_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pokebattle")
        var COMMAND_POKEBATTLE_PERMISSION_LEVEL: Int = 3

        @SerializedName("command.battle")
        var COMMAND_BATTLE_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.compdelete")
        var COMMAND_COMPDELETE_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pokedelete")
        var COMMAND_POKEDELETE_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pokeivs")
        var COMMAND_POKEIVS_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.emptybox")
        var COMMAND_EMPTYBOX_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pokeshoutall")
        var COMMAND_POKESHOUT_ALL_PERMISSION_LEVEL: Int = 2

        @SerializedName("command.pokeodds")
        var COMMAND_POKEODDS_PERMISSION_LEVEL: Int = 3

        @SerializedName("command.pokekill")
        var COMMAND_POKEKILL_PERMISSION_LEVEL: Int = 3

        @SerializedName("command.givepoketoken")
        var COMMAND_GIVE_POKETOKEN_PERMISSION: Int = 3

        @SerializedName("command.giveshinyball")
        var COMMAND_GIVE_SHINYBALL_PERMISSION: Int = 3

        @SerializedName("command.battlespectate")
        var COMMAND_BATTLESPECTATE_PERMISSION: Int = 2
    }

    inner class ShowcaseConfig {
        var isShowcaseEnabled: Boolean = true
        var apiSecret: String = "To start using showcase, please goto https://cobblemonextras.com/showcase"

        var syncIntervalMinutes: Int = 5
        var debug: Boolean = false
        var async: Boolean = true
    }

    inner class CustomModelsConfig {
        var SHINY_TOKEN = 100;
        var NATURE_TOKEN = 101;
        var IV_TOKEN = 102;
        var EV_TOKEN = 103;

        // PokeBall Item
        var SHINY_BALL = 100;
    }

    companion object {
        var GSON: Gson = GsonBuilder()
                .disableHtmlEscaping()
                .setPrettyPrinting()
                .create()
    }
}
